### PR TITLE
cmake: installing process rework, added warnings to GNU GCC compiler.

### PR DESCRIPTION
**Pull request by ivanstepanovftw**
cmake: installing process rework, added warnings to GNU GCC compiler.

- Static and dynamic library build disabled by default because all headers covered by tests and targets are not used anywhere. Code inspection works without these targets. Solves #433.
- Added warnings to GNU GCC compiler.
- Moved headers to `./include/glm` and sources to `./src`, solves issue when `CMakeLists.txt` and `glm.cpp` accidentally in `$PREFIX/include` folder after installing.
- Removed `glmTargets.cmake` for installation as unnecessary, solves #810.
- Replaced `glmConfig.cmake.in` and `glmBuildConfig.cmake.in` with CMake builtin routines, solves #931.
- GLM's test targets are no more imported to client's application.
- GLM should be added via namespace now, breaks backward-capability for CMake (replace `glm` with `glm::glm`), solves #748.

### CODE REVIEW

**Comment by ivanstepanovftw** [comment]
@Groovounet, ready for review

**Comment by ivanstepanovftw** [comment]
`find_package` works fine: _(fix #931)_
```
find_package(glm 0.9.8 REQUIRED)
...
add_executable(scratch scratch.cc)
target_link_libraries(scratch glm::glm)
```

FetchContent works fine: _(control)_
```cmake
FetchContent_Declare(glm
        # GIT_REPOSITORY https://github.com/g-truc/glm.git
        GIT_REPOSITORY https://github.com/ivanstepanovftw/glm.git
        GIT_TAG        cmake_revision
        GIT_SHALLOW TRUE
        )

FetchContent_GetProperties(glm)
if(NOT glm_POPULATED)
    FetchContent_Populate(glm)
    add_subdirectory(${glm_SOURCE_DIR} ${glm_BINARY_DIR})
endif()
...
add_executable(scratch scratch.cc)
target_link_libraries(scratch glm::glm)
```

File: test/ext/CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,9 @@ glmCreateTestGTC(ext_scalar_common)
 glmCreateTestGTC(ext_scalar_constants)
 glmCreateTestGTC(ext_scalar_int_sized)
 glmCreateTestGTC(ext_scalar_uint_sized)
-glmCreateTestGTC(ext_scalar_integer)
+if (NOT CMAKE_CXX_STANDARD STREQUAL 98)
```
Comment (ivanstepanovftw) [inline test/ext/CMakeLists.txt]: This check should be removed

Comment (ivanstepanovftw) [reply test/ext/CMakeLists.txt]: Fixed 5 days ago

**Comment by Groovounet** [comment]
Moving the files will break a lot of projects using GLM directly from the repo so this is not an option.

There is probably a lot to do regarding the CMake install but I am actually considering removing all this as it's a constant source of trouble..

Thanks for contributing,
Christophe

**Pull request closed without merging**